rust_src/src/wrterm.rs

```diff
@@ -0,0 +1,182 @@
+//! wrterm.rs
+
+use std::ptr;
+
+use remacs_macros::lisp_fn;
+
+use crate::{
+    frame::LispFrameRef,
+    lisp::{ExternalPtr, LispObject},
+    remacs_sys::{
+        font, image, wr_display_info, wr_output, Display, Emacs_Pixmap, Pixmap, Qnil, WRImage,
+        Window, XrmDatabase,
+    },
+};
+
+#[repr(transparent)]
+pub struct DisplayInfoRef(*mut wr_display_info);
+unsafe impl Sync for DisplayInfoRef {}
+
+pub type OutputRef = ExternalPtr<wr_output>;
+pub type DisplayRef = ExternalPtr<Display>;
+pub type ImageRef = ExternalPtr<WRImage>;
+
+#[no_mangle]
+pub static tip_frame: LispObject = Qnil;
+
+#[no_mangle]
+pub static wr_display_list: DisplayInfoRef = DisplayInfoRef(ptr::null_mut());
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_fontset(output: OutputRef) -> i32 {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_font(output: OutputRef) -> *const font {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_window_desc(output: OutputRef) -> Window {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_display_info(output: OutputRef) -> DisplayInfoRef {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_display(display_info: DisplayInfoRef) -> DisplayRef {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_baseline_offset(output: OutputRef) -> i32 {
+    unimplemented!();
+}
+
+#[allow(unused_variables)]
+#[no_mangle]
+pub extern "C" fn wr_get_pixel(ximg: ImageRef, x: i32, y: i32) -> i32 {
```

Comment:
For these extern "C" functions, is it intentional to type the arguments as i32 vs a type from libc (e.g. libc::c_int)?

Reply:
Why I use "i32" instead of "c_int" here is to avoid `as` casts in function body. IMHO, `libc::c_int` equals `i32`.

Reply:
I remember that using `i32` caused issues in certain situations, but I don't remember why.